src/atmospheric_thermodynamics.jl

```diff
@@ -126,10 +128,10 @@ struct AtmosphereThermodynamics{FT, S, C, F}
     deposition :: F
 end
 
-Base.eltype(::AtmosphereThermodynamics{FT}) where FT = FT
+Base.eltype(::AtmosphereThermodynamics{FT, S, C, F}) where {FT, S, C, F} = FT
```

Comment:
```suggestion
Base.eltype(::AtmosphereThermodynamics{FT}) where FT = FT
```

the additional parameters are not needed (following parameters are implicit)

Reply:
yeah... it was part of the attempts to find where Any was coming from